### PR TITLE
Improved heatmap display for confusion matrix report

### LINKED ISSUE #2427: [FEAT] improve confusion matrix display
1. the percentages looks weird (should sum to 100)
2. add a column containing info of the accuracy per class

### PR DESCRIPTION
**Pull request by harsh-deepchecks**
#### Reference Issues/PRs

Fixes #2427 and [DEE-439 - "[FEAT] improve confusion matrix display" (deepchecks/deepchecks #2427)](https://linear.app/deepchecks/issue/DEE-439/[feat]-improve-confusion-matrix-display-deepchecksdeepchecks-2427)

#### What does this implement/fix? Explain your changes.
- Modified the heatmap to have each cell separated by a small gap so that user is able to distinguish between each cell as earlier if the cells had same colour, they were connected.
- Added hover mechanism with three data information:
    - % of data points in each cell across all the data points.
    - % of data point in each cell across the row.
    - % of data point in each cell across the column.

#### Any other comments?
Below are the heatmaps for different types of data:
- NLP
     -  image
 
- Vision
    -  image

- Tabular  
    -  image
 

---
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/deepchecks/deepchecks/blob/main/CONTRIBUTING.rst

### CODE REVIEW

File: deepchecks/utils/abstracts/confusion_matrix_abstract.py [hunk 1]
```diff
@@ -60,29 +63,53 @@ def create_confusion_matrix_figure(confusion_matrix_data: np.ndarray, classes_na
...
-        x=classes_names, y=classes_names, z=z,
-        text=confusion_matrix_data, texttemplate=text_template))
-    fig.data[0].colorbar.title = color_bar_title
-    fig.update_layout(title=plot_title)
+
+    display = []
+    accuracy_array = np.diag(confusion_matrix_norm).round(decimals=2)
+    sorted_map = dict(sorted(dict(enumerate(accuracy_array)).items(), key=lambda item: item[1]))
+
+    worst_class_indices = [index for index, acc in sorted_map.items() if acc < MIN_ACCURACY_FOR_GOOD_CLASSES]
```
Comment (Nadav-Barak) [inline deepchecks/utils/abstracts/confusion_matrix_abstract.py:77]: would remove the if and the MIN_ACCURACY_FOR_GOOD_CLASSES variable

File: deepchecks/utils/abstracts/confusion_matrix_abstract.py [hunk 2]
```diff
@@ -60,29 +63,53 @@ def create_confusion_matrix_figure(confusion_matrix_data: np.ndarray, classes_na
...
+
+    display_msg = textwrap.dedent(
+            f'The overall accuracy of your model is: {round(np.sum(accuracy_array)/len(accuracy_array), 2)}%.<br>'
+            f'Best accuracy achieved on samples with <b>{classes_names[np.argmax(accuracy_array)]}</b> '
+            f'label ({np.max(accuracy_array)}%).'
+        )
+
+    if len(worst_class_indices) == 0:
```
Comment (Nadav-Barak) [inline deepchecks/utils/abstracts/confusion_matrix_abstract.py:85]: if all classes have 100 % no need for the sentence about best class as well

File: deepchecks/utils/abstracts/confusion_matrix_abstract.py [hunk 3]
```diff
@@ -9,6 +9,7 @@
 # ----------------------------------------------------------------------------
 #
 """The confusion_matrix_report check module."""
+import textwrap
```
Comment (Nadav-Barak) [inline deepchecks/utils/abstracts/confusion_matrix_abstract.py:12]: What is this used for?
Try to see if you can work without it (we try to limit as much as possible number of external packages)

**Review by Nadav-Barak — APPROVED** [review]

File: deepchecks/utils/abstracts/confusion_matrix_abstract.py [hunk 4]
```diff
@@ -60,29 +63,53 @@ def create_confusion_matrix_figure(confusion_matrix_data: np.ndarray, classes_na
...
+                              np.finfo(float).eps) * 100).round(decimals=2)
+    custom_hoverdata = np.dstack((percent_data_each_cell, percent_data_each_row, percent_data_each_col))
+
+    fig = go.Figure(data=go.Heatmap(x=classes_names, y=classes_names, z=z, customdata=custom_hoverdata,
+                                    xgap=1, ygap=1, text=confusion_matrix_data, texttemplate='%{text}',
+                                    hovertemplate='% out of all data: <b>%{customdata[0]}%</b><br>% out '
+                                                  'of row: <b>%{customdata[1]}%</b><br>% out of column: '
+                                                  '<b>%{customdata[2]}%</b><extra></extra>',
+                                    showscale=False))
+    fig.update_layout(title='Confusion matrix', title_x=0.5)
```
Comment (Nadav-Barak) [inline deepchecks/utils/abstracts/confusion_matrix_abstract.py:107]:
```suggestion
    fig.update_layout(title='Confusion Matrix (# Samples)', title_x=0.5)
```

Comment (harsh-deepchecks) [reply deepchecks/utils/abstracts/confusion_matrix_abstract.py]: Done

File: deepchecks/utils/abstracts/confusion_matrix_abstract.py (hunk 3 above)
Comment (harsh-deepchecks) [reply deepchecks/utils/abstracts/confusion_matrix_abstract.py]: Done

File: deepchecks/utils/abstracts/confusion_matrix_abstract.py (hunk 2 above)
Comment (harsh-deepchecks) [reply deepchecks/utils/abstracts/confusion_matrix_abstract.py]: Done

File: deepchecks/utils/abstracts/confusion_matrix_abstract.py (hunk 1 above)
Comment (harsh-deepchecks) [reply deepchecks/utils/abstracts/confusion_matrix_abstract.py]: Done

**Pull request merged**